### DISCUSSION TITLE
What's the best way to fix `'blink-cmp-signature' is not a valid language name`?

### DISCUSSION THREAD

**Discussion by tristan957** [saghen/blink.cmp#776] [General]
It's a treesitter error, and I don't see a parser in nvim-treesitter repo.

```
Error executing vim.schedule lua callback: /usr/share/nvim/runtime/lua/vim/treesitter/language.lua:101: 'blink-
cmp-signature' is not a valid language name
```

**Comment by Quitlox** [comment]
This is (likely) not a `blink.cmp` problem. You have some other plugin that is trying to parse the signature window using treesitter. You'll need to disable plugins until you find the culprit, and then ensure it does not activate on the `blink-cmp-signature` filetype.

**Reply by tristan957** [reply]
Awesome, thank you for that tip. I'll give it a go and report back what I find.

**Reply by tristan957** [reply]
Here is the full backtrace:

```
Error executing vim.schedule lua callback: /usr/share/nvim/runtime/lua/vim/treesitter/language.lua:101: 'blink-
cmp-signature' is not a valid language name
stack traceback:
        [C]: in function 'error'
        /usr/share/nvim/runtime/lua/vim/treesitter/language.lua:101: in function 'add'
        /usr/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:111: in function 'new'
        /usr/share/nvim/runtime/lua/vim/treesitter.lua:41: in function '_create_parser'
        /usr/share/nvim/runtime/lua/vim/treesitter.lua:108: in function 'get_parser'
        /usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:117: in function 'compute_folds_levels'
        /usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:380: in function 'fn'
        /usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:293: in function </usr/share/nvim/runtime/lua/vim/
treesitter/_fold.lua:289>
```

It indicates to me that this is coming from Neovim itself. Maybe I'm wrong though.

**Reply by Quitlox** [reply]
Maybe look for a fold related plugin as the trace suggest and see if the issue persists after disabling those plugins.

It's very unlikely this is caused by Neovim itself. My guess is again that some plugin is triggering Neovim to attempt to parse/fold the window with filetype `blink-cmp-signature` opened by `blink.cmp`. This is a problem of said plugin (or some custom configuration you may have).

**Reply by Quitlox** [reply]
Only way to find the problem is unfortunately to trim down your configuration until it doesn't happen anymore. Either you'll stumble on an incompatible / misbehaving plugin, or you found a bug in `blink.cmp`.

**Reply by tristan957** [reply]
My `foldexpr` is set to

```
vim.opt.foldexpr = "v:lua.vim.treesitter.foldexpr()"
```

That sounds like the problem. I'll create a filetype autocommand to set foldexpr to something else.

**Reply by husa** [reply]
@tristan957 have you found a fix?

**Reply by tristan957** [reply]
The problem was caused by another plugin https://github.com/nvim-treesitter/nvim-treesitter-context. It got an update like a month ago-ish which fixed the issue.

**Reply by husa** [reply]
Thanks for update

**Comment by Muizzyranking** [comment]
I also had this problem but it stopped when I set signature to false.

The issues happens when I select copilot's suggestion so I am guessing it is from [giuxtaposition/blink-cmp-copilot](https://github.com/giuxtaposition/blink-cmp-copilot.git).

**Reply by tristan957** [reply]
I don't use that plugin, so I at least don't think it is specific to it

**Reply by Muizzyranking** [reply]
Maybe you should turn off signature.

**Reply by tristan957** [reply]
Yes, the issue goes away without signature. Was hoping to get a better solution than that.

**Comment by tristan957** [comment]
I keep seeing this in Zig files, so I tried to narrow down a problematic plugin. I could reproduce the issue with just these plugins:

![image](https://github.com/user-attachments/assets/6bdb2977-7911-4f64-a0e5-50cc6b682c52)

With this set of plugins and my pared down config, I could only reproduce the issue if I had the following settings:

```lua
        vim.wo.foldmethod = "expr"
        vim.wo.foldexpr = "v:lua.vim.treesitter.foldexpr()"
```

I still find this super confusing... If anyone has more guidance, that would be awesome
